src/line_info.c

```diff
@@ -349,11 +349,11 @@ static int print_wrap(WINDOW *win, struct line_info *line, int max_x, int max_y)
  */
 static uint16_t line_info_add_msg(wchar_t *buf, size_t buf_size, const char *msg)
 {
-    if (msg == NULL || msg[0] == '\0') {
+    if (msg == NULL || msg[0] == '\0' || buf_size < sizeof(wchar_t)) {
```

Comment:
Does this mean buf_size is measured in bytes, not in elements? If so, that's very unusual and should be documented.

Reply:
Do you think this function should require the number of elements instead of doing the division during the call to `mbs_to_wcs_buf()` at line 356?

Reply:
Yes, this makes a lot more sense. Now you don't need the sizeof check here, right?